docs/examples/edit_distance.md

```diff
@@ -0,0 +1,121 @@
+# Edit Distance
+
+- Given two strings `str1` and `str2`, what is the minimum number of edits
+  to convert `str1` to `str2`?
+- Each edit costs 1, and is either an add, delete, or replace.
+- Goal: find a minimum cost set of edits that converts `str1` to `str2`.
+
+## Dynamic Programming Solution
+
+> A fully executable example can be found on our [GitHub](https://github.com/itsdawei/dpvis/tree/main/demos/edit_distance_2d.py).
+
+Let $OPT[i, j]$ denote the cheapest conversion between the first $i$ characters of `str1`
+and the first $j$ characters of `str2`.
+
+**BASE CASES: $i$ or $j$ is $0$.** At least one of our strings is empty, 
+so we should just pay the cost to remove the remaining letters in the other string. So
+we should record that $OPT[i, j] = max(i, j)$.
+
+**CASE 1: `str1[i] == str2[j]`.** The last letter of `str1` is the same as
+the last letter of `str2`. So we should leave the last letters alone and convert
+`str1[i-1]` to `str2[j-1]` as cheaply as possible. So, $OPT[i, j] = OPT[i-1, j-1]$.
+
+**CASE 2: `str1[i] != str2[j]`.** The last letter of `str1` is not the same as
+the last letter of `str2`. At this point, we have to either add the last letter of 
+`str2` to `str1`, delete the last letter of `str1`, or replace the last letter of 
+`str1` with the last letter of `str2`. Each option is an edit, so we must record
+$OPT[i, j] = 1 + min(OPT[i, j - 1], OPT[i - 1, j], OPT[i - 1, j - 1])$.
+
+## Visualization with `dpvis`
+
+We can visualize this with `dpvis` as follows:
+
+```python linenums="1"  hl_lines="1 17 60-62"
+from dp import DPArray, display
+
+def edit_distance(str1, str2):
+    """
+    Edit Distance Problem:
+    Given two strings str1 and str2 of lengths m and n, respectively, what is 
+    the cost of the cheapest set of actions that converts str1 into str2. 
+    The following actions are possible:
+    add before/after index i, remove before/after index i, replace at index i
+
+    Solution adapted from Bhavya Jain's solution:
+    https://www.geeksforgeeks.org/edit-distance-dp-5/
+    """
+    m = len(str1)
+    n = len(str2)
+    # Initialize an m+1 x n+1 array
+    OPT = DPArray((m + 1, n + 1), array_name="Edit Distance", dtype=int)
+
+    # Base cases: either str1 or str2 is empty
+    # Then we have to pay to remove/add the remaining letters
+    for i in range(m + 1):
+        OPT[i, 0] = i
+    for j in range(n + 1):
+        OPT[0, j] = j
+    OPT.annotate("Base cases: no remaining letters in str1 or str2.")
```

Comment:
I don't think this annotation is very informative unless you already understand what is going on in the algorithm. I think maybe put the formula in line 55 and line 57 here.